crates/sui/src/client_commands.rs

```diff
@@ -665,6 +678,24 @@ impl SuiClientCommands {
         context: &mut WalletContext,
     ) -> Result<SuiClientCommandResult, anyhow::Error> {
         let ret = Ok(match self {
+            SuiClientCommands::ProfileTransaction { rpc_url, tx_digest } => {
+                let key = "MOVE_VM_PROFILE";
```

Comment:
We shouldn't have to use environment variables to control things that are internal to this tool/process -- can we control this by creating a configuration flag that we pass in directly, instead?

Also, will profiling work on all builds of the CLI?  I've noticed that some of these things that are currently controlled by environment variables are only enabled in debug builds, and we should make sure that the version of the VM that's built with the CLI can always (even in release) be built with these tools.

Reply:
see https://github.com/MystenLabs/sui/pull/14648 for removing restriction on debug builds only for the profiler

@amnn great point here! The original profiler relies on an environment variable being set and that is hard / messy / maybe not possible to set programmatically for anything but the current process. I like the idea of using a config to enable the profiling instead. I looked into that and it seems like that would take quite a lot of plumbing to pass a config value to the place where the profiler is invoked within the vm.
The easiest path forward would be to say via documentation that you can set this environment variable when calling this existing replay command and it will output a profile that can be opened in a flamegraph tool, and that's the profile story. What are your thoughts on this, vs potentially doing the work to make a standalone command that abstracts those details out. Maybe there is an easier way? cc: @oxade

Reply:
Personally, I think our job here is to make things easier for the user, and while the plumbing work is annoying, it's the right thing to do for a production tool (I wouldn't care as much if we were talking about an internal tool).  I'm also curious to get @tzakian's thoughts on the work involved here as I believe he is the originator of the profiler and may have some tricks up his sleeve 🤞 .

Reply:
I am the originator of this pattern (perpetrator maybe more apt though). This started out with the VM-tracing work long ago to gather test coverage results (and then the debugger, and then when @oxade went to implement it I was like...just use the current design where we debug-build flag it and then set the env variable to turn it on...).

I think as long as these were internal only this was fine but agree with Ashok that adding a general purpose flag to turn on debugging, or gas profiling would make sense to me.

One possible way to lessen the plumbing here (but sadly not remove it completely): the interpreter already has a `VMRuntimeLimitsConfig`, that could be change to a more general purpose `VMRuntimeConfig` or the like and then the profiler/other configs could be added to that possibly? That should be plumbed pretty much down to where it's needed, but there might be a little bit of hooking up needed on either end of things.